cassandane/tiny-tests/Replication/mboxgroups

```diff
@@ -0,0 +1,116 @@
+#!perl
+use Cassandane::Tiny;
+
+sub test_mboxgroups
+    :needs_component_replication :Mboxgroups :ReverseACLs
+{
+    my ($self) = @_;
+
+    my $user = 'brandnew';
+    $self->{instance}->create_user($user);
+
+    my $mastersvc = $self->{instance}->get_service('imap');
+    my $masterstore = $mastersvc->create_store(username => $user);
+    my $mastertalk = $masterstore->get_client();
+
+    my $adminstore = $mastersvc->create_store(username => 'admin');
+    my $admintalk = $adminstore->get_client();
+
+    $admintalk->_imap_cmd('SETUSERGROUP', 0, '', 'cassandane', 'group:shared');
+    $admintalk->_imap_cmd('SETUSERGROUP', 0, '', 'brandnew', 'group:shared');
+
+    $admintalk->setacl("user.cassandane", "group:shared", "lrs");
+
+    $mastertalk->create("INBOX.Test") || die;
+    $mastertalk->create("INBOX.Test.Sub") || die;
+    $mastertalk->create("INBOX.Test Foo") || die;
+
+    my $ldata = $mastertalk->list("", "*");
+    $self->assert_deep_equals($ldata, [
+          [
+            [
```

Comment:
You could use `setup_mailbox_structure()` and `assert_mailbox_structure()` here to remove a lot of clutter.  See the List tests for lots of examples.